tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/SpecWorkflowTool/SpecWorkFlowTool.cs

```diff
@@ -56,6 +61,136 @@ public class SpecWorkflowTool(IGitHubService githubService,
+                {
+                    var namespaceApprovalIssueURL = releasePlan?.NamespaceApprovalIssueURL;
+                    if (string.IsNullOrEmpty(namespaceApprovalIssueURL))
+                    {
+                        response.Status = "Failed";
+                        response.Details.Add("Namespace approval issue is required to verify namespace approval status. Provide the namespace approval issue URL and link it to the release plan.");
+                        return response;
+                    }
+                    
+                    var match = NameSpaceIssueUrlRegex().Match(namespaceApprovalIssueURL);
+                    // Check if the namespace approval issue is a valid GitHub issue number
+                    if (!match.Success)
+                    {
+                        response.Status = "Failed";
+                        response.Details.Add($"Invalid namespace approval issue '{namespaceApprovalIssueURL}'. It should be a valid GitHub issue in Azure/azure-sdk repo.");
+                        return response;
+                    }
+
+                     // Get issue number from the match
+                    var issueNumber = int.Parse(match.Groups[1].Value);
+                    var issue = await githubService.GetIssueAsync(namespaceApprovalRepoOwner, namespaceApprovalRepoName, issueNumber);
+                    if(issue == null)
+                    {
+                        response.Status = "Failed";
+                        response.Details.Add($"Failed to verify approval status. Namespace approval issue #{namespaceApprovalIssueURL} not found in {namespaceApprovalRepoOwner}/{namespaceApprovalRepoName}.");
+                        return response;
+                    }
+
+                    // Verify if issue has label 'mgmt-namespace-review'
+                    if (!issue.Labels.Any(label => label.Name.Equals("mgmt-namespace-review", StringComparison.OrdinalIgnoreCase)))
+                    {
+                        response.Status = "Failed";
+                        response.Details.Add($"Namespace approval issue {namespaceApprovalIssueURL} does not have the required 'mgmt-namespace-review' label.");
+                        return response;
+                    }
+
+                    if (issue.State == ItemState.Open)
+                    {
+                        response.Status = "Failed";
+                        response.Details.Add($"Namespace approval is required for first preview release of SDK. Provide the namespace approval issue URL and link it to the release plan");
+                        return response;
+                    }
+
+                }
+                response.Status = "Success";
+                response.Details.Add($"Namespace approval check completed: either approval is not required for this SDK release, or it has already been successfully linked to the release plan.");
+                return response;
+            }
+            catch (Exception ex)
```

Comment:
I'd say you can remove this whole try/catch since it's already handled in the outer method that calls this. It will simplify the code a bit and also preserve the stack trace.